### PR TITLE
Shm15_1-4_kozlofski - tasks from 18.06 online lesson

### PR DESCRIPTION
**Pull request by kozlofski**
I also made classes Ship and Cargo, that is tasks SHM1-4

### CODE REVIEW

File: shm/fruit.hpp [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    }
+
+    size_t getPrice() const override {
+        return basePrice_ * (timeToExpire_ / expiryDate_);
+    }
+
+private:
+    size_t timeToExpire_;
+    const size_t expiryDate_{10};
+};
```
Comment (kyrtaq) [inline shm/fruit.hpp:27]: add a new line

File: shm/item.hpp [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        return basePrice_;
+    }
+
+    size_t getPrice() const override {
+        return basePrice_ * static_cast<int>(rarity_);
+    }
+
+private:
+    Rarity rarity_;
+};
```
Comment (kyrtaq) [inline shm/item.hpp:31]: new line

File: shm/alcohol.hpp [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    size_t getAmount() const override {
+        return amount_;
+    }
+
+    size_t getBasePrice() const override {
+        return basePrice_;
+    }
+
+    size_t getPrice() const override {
+        return basePrice_ * (alcohol / 96);
```
Comment (NG90) [inline shm/alcohol.hpp:19]: Magic number. Create const variable defining `maxAlcoPower` within the class.

File: shm/fruit.hpp (hunk 1 above)
Comment (NG90) [inline shm/fruit.hpp:27]: new line

File: shm/ship.cpp [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    crew_ -= crew;
+    return *this;
+}
+
+void Ship::load(const std::shared_ptr<Cargo>& cargo) {
+    cargo_.push_back(cargo);
+    // apply checking if this cargo type exist
+    // and have same expiry date!
+    // if yes - increase amount
+    // else pushback
```
Comment (NG90) [inline shm/ship.cpp:40]: will this check be implemented in another task? If not, maybe it should be done within this task.

File: shm/ship.cpp [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+void Ship::unload(Cargo* cargo) {
+    // this should remove given amount of cargo
```
Comment (NG90) [inline shm/ship.cpp:44]: will it be implemented in another task? If not, maybe it should be done within this task.

**Pull request merged**